src/app/test_executive/epoch_ledger_medium_bootstrap.ml

```diff
@@ -0,0 +1,147 @@
+open Core
+open Async
+open Integration_test_lib
+
+module Make (Inputs : Intf.Test.Inputs_intf) = struct
+  open Inputs
+  open Engine
+  open Dsl
+
+  open Test_common.Make (Inputs)
+
+  (* TODO: find a way to avoid this type alias (first class module signatures restrictions make this tricky) *)
+  type network = Network.t
+
+  type node = Network.Node.t
+
+  type dsl = Dsl.t
+
+  let config =
+    let k = 2 in
+
+    let open Test_config in
+    { default with
+      k
+    ; slots_per_epoch = 3 * 8 * k
+    ; requires_graphql = true
+    ; genesis_ledger =
+        [ { account_name = "node-a-key"; balance = "1000"; timing = Untimed }
+        ; { account_name = "node-b-key"; balance = "1000"; timing = Untimed }
+        ; { account_name = "node-c-key"; balance = "0"; timing = Untimed }
+        ]
+    ; block_producers =
+        [ { node_name = "node-a"; account_name = "node-a-key" }
+        ; { node_name = "node-b"; account_name = "node-b-key" }
+        ; { node_name = "node-c"; account_name = "node-c-key" }
+        ]
+    ; proof_config =
+        { proof_config_default with
+          work_delay = Some 1
+        ; transaction_capacity =
+            Some Runtime_config.Proof_keys.Transaction_capacity.medium
+        }
+    }
+
+  let transactions_sent = ref 0
+
+  let num_proofs = 2
+
+  let padding_payments () =
+    let needed_for_padding =
+      Test_config.transactions_needed_for_ledger_proofs config ~num_proofs
+    in
+    if !transactions_sent >= needed_for_padding then 0
+    else needed_for_padding - !transactions_sent
+
+  let send_padding_transactions ~fee ~logger ~n nodes =
```

Comment:
I think that this is duplication of zkapp_nonce_test. Do we plan to move it to test_common?